### PR TITLE
Update transaction dry run implementation

### LINKED ISSUE #1388: Update transaction dry run implementation
Description
The `txpool_dryRunTransaction` endpoint from SDK returns encoded event data. The implementation in Lisk Service needs to be updated to support dynamic events decoding.

Acceptance Criteria
- The dryRunTransaction implementation in `blockchain-connector` is updated to decode events dynamically based on module and command name.
- Ensure events data in response for `txpool_dryRunTransaction` is decoded
- Integration tests are up-to-date

Additional Information

### PR DESCRIPTION
**Pull request by priojeetpriyom**
### What was the problem?
This PR resolves #1388

### How was it solved?

- [x] The dryRunTransaction implementation in blockchain-connector is updated to decode events dynamically based on module and command name.
- [x] Ensure events data in response for txpool_dryRunTransaction is decoded
- [x] Integration tests are up-to-date

### How was it tested?
- [x] Local

### CODE REVIEW

File: tests/schemas/api_v3/transactionsDryRun.schema.js [hunk 1]
```diff
@@ -24,20 +24,32 @@ const event = {
...
 };
-
-const goodRequestSchema = {
-	data: Joi.object(dryrunTransactionResponseSchema).required(),
-	meta: Joi.object().required(),
+const dryrunTransactionFailedResponseSchema = {
+	result: Joi.number().integer().valid(1).required(),
+	events: Joi.array().items(Joi.object(event).optional()).required(),
+};
+const dryrunTransactionInvalidResponseSchema = {
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionsDryRun.schema.js:38]: Please pay attention to the changes in the suggestion.
```suggestion

const dryrunTransactionFailResponseSchema = {
	result: Joi.number().integer().valid(0).required(),
	events: Joi.array().items(Joi.object(event).optional()).required(),
};

const dryrunTransactionInvalidResponseSchema = {
```

File: tests/schemas/api_v3/transactionsDryRun.schema.js [hunk 2]
```diff
@@ -24,20 +24,32 @@ const event = {
...
 };
 
 module.exports = {
-	goodRequestSchema: Joi.object(goodRequestSchema).required(),
-	dryrunTransactionResponseSchema: Joi.object(dryrunTransactionResponseSchema).required(),
+	dryrunTransactionSuccessResponseSchema: Joi.object(
+		dryrunTransactionSuccessResponseSchema,
+	).required(),
+	dryrunTransactionFailedResponseSchema: Joi.object(
+		dryrunTransactionFailedResponseSchema,
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionsDryRun.schema.js:49]: Please adjust the tests accordingly.
```suggestion
	dryrunTransactionFailResponseSchema: Joi.object(
		dryrunTransactionFailResponseSchema,
```

File: tests/integration/api_v3/http/transactionsDryRun.test.js [hunk 3]
```diff
@@ -48,9 +49,8 @@ describe('Post dryrun transactions API', () => {
 		);
 
 		expect(response).toMap(goodRequestSchema);
-		expect(response.data).toMap(dryrunTransactionResponseSchema);
+		expect(response.data).toMap(dryrunTransactionSuccessResponseSchema);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/http/transactionsDryRun.test.js:52]: Please update all the test descriptions (wherever necessary) to be more accurate, explaining what is being tested.

File: tests/integration/api_v3/rpc/transactionsDryRun.test.js [hunk 4]
```diff
@@ -55,7 +57,7 @@ describe('Method post.transactions.dryrun', () => {
 		const { result } = response;
 		expect(result).toMap(goodRequestSchema);
 		expect(result.data).toBeInstanceOf(Object);
-		expect(result.data).toMap(dryrunTransactionResponseSchema);
+		expect(result.data).toMap(dryrunTransactionSuccessResponseSchema);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/transactionsDryRun.test.js:60]: Same as above.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by nagdahimanshu — APPROVED** [review]
LGTM 👍

File: tests/schemas/api_v3/transactionsDryRun.schema.js [hunk 5]
```diff
@@ -24,20 +24,32 @@ const event = {
...
 	topics: Joi.array().items(Joi.string().pattern(regex.TOPIC)).required(),
 	height: Joi.number().integer().min(0).required(),
+	id: Joi.string().pattern(regex.HEX).required(),
 };
 
-const dryrunTransactionResponseSchema = {
-	success: Joi.boolean().required(),
+const dryrunTransactionSuccessResponseSchema = {
+	result: Joi.number().integer().valid(1).required(),
 	events: Joi.array().items(Joi.object(event).optional()).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionsDryRun.schema.js:32]:
```suggestion
	events: Joi.array().items(Joi.object(event).required()).required(),
```

File: tests/schemas/api_v3/transactionsDryRun.schema.js [hunk 6]
```diff
@@ -24,20 +24,32 @@ const event = {
...
 };
-
-const goodRequestSchema = {
-	data: Joi.object(dryrunTransactionResponseSchema).required(),
-	meta: Joi.object().required(),
+const dryrunTransactionFailResponseSchema = {
+	result: Joi.number().integer().valid(0).required(),
+	events: Joi.array().items(Joi.object(event).optional()).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionsDryRun.schema.js:36]:
```suggestion
	events: Joi.array().items(Joi.object(event).required()).required(),
```

File: tests/schemas/api_v3/transactionsDryRun.schema.js [hunk 7]
```diff
@@ -24,20 +24,32 @@ const event = {
...
+};
+const dryrunTransactionInvalidResponseSchema = {
+	result: Joi.number().integer().valid(-1).required(),
+	events: Joi.array().items(Joi.object(event).optional()).required(),
```
Comment (sameersubudhi) [inline tests/schemas/api_v3/transactionsDryRun.schema.js:40]:
```suggestion
	events: Joi.array().length(0).required(),
```

File: tests/integration/api_v3/rpc/transactionsDryRun.test.js [hunk 8]
```diff
@@ -34,16 +35,35 @@ const {
...
 const postDryrunTransaction = async params => request(wsRpcUrl, 'post.transactions.dryrun', params);
 const postTransaction = async params => request(wsRpcUrl, 'post.transactions', params);
 
 describe('Method post.transactions.dryrun', () => {
+	it('Returns proper response for transaction with less than required fee', async () => {
+		const response = await postDryrunTransaction(
+			{
+				transaction: TRANSACTION_OBJECT_FAIL,
+			},
+		);
```
Comment (sameersubudhi) [inline tests/integration/api_v3/rpc/transactionsDryRun.test.js:56]: Can be a one-liner.
```suggestion
		const response = await postDryrunTransaction({ transaction: TRANSACTION_OBJECT_FAIL });
```

File: tests/integration/api_v3/constants/transactionsDryRun.js [hunk 9]
```diff
@@ -51,10 +50,29 @@ const TRANSACTION_OBJECT_INVALID = {
...
+	params: {
+		recipientAddress: 'lskjokybqso65y599cbschhzmguoas2uzq2jnnbo2',
+		amount: '10000000000',
+		tokenID: '0400000000000000',
+		data: '',
+	},
+	id: 'dbfc75ea10fc3297213a104079faaa0fc983e3cf0ad05044d5f9d168ff83b172',
+};
+
+const TRANSACTION_ENCODED_VALID = '0a05746f6b656e12087472616e7366657218002080c2d72f2a203972849f2ab66376a68671c10a00e8b8b67d880434cc65b04c6ed886dfa91c2c322f0a0804000000000000001080c8afa0251a1402604d9e57a39772fa12f2a860ecf6c1e9cae9112207546573742074783a40d1ccc2d5fd0cd953b77c608102b7dbb6df7748474667af9e11b6aa050aa56fdc4758b9ae8fad1e0c01c6cc7d5967311e21710c53e0369e711057c09741d75804';
```
Comment (sameersubudhi) [inline tests/integration/api_v3/constants/transactionsDryRun.js:71]: Please add test cases for `TRANSACTION_ENCODED_INVALID` as well. And, adjust other transactions to implement the exact params as returned in the metadata schema corresponding the module/command, .i.e. remove `accountInitializationFee`.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

File: tests/integration/api_v3/constants/transactionsDryRun.js (hunk 9 above)
Comment (priojeetpriyom) [reply tests/integration/api_v3/constants/transactionsDryRun.js:73]: Applied.

File: tests/schemas/api_v3/transactionsDryRun.schema.js (hunk 5 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionsDryRun.schema.js]: Applied.

File: tests/schemas/api_v3/transactionsDryRun.schema.js (hunk 6 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionsDryRun.schema.js]: Applied.

File: tests/schemas/api_v3/transactionsDryRun.schema.js (hunk 7 above)
Comment (priojeetpriyom) [reply tests/schemas/api_v3/transactionsDryRun.schema.js]: Applied.

**Review by sameersubudhi — APPROVED** [review]

**Pull request merged**